models/Asset.php

```diff
@@ -709,7 +710,14 @@ public function correctPath()
         if (Asset\Service::pathExists($this->getRealFullPath())) {
             $duplicate = Asset::getByPath($this->getRealFullPath());
             if ($duplicate instanceof Asset && $duplicate->getId() != $this->getId()) {
-                throw new \Exception('Duplicate full path [ ' . $this->getRealFullPath() . ' ] - cannot save asset');
+                $duplicateFullPathException = new DuplicateFullPathException(
+                    'Duplicate full path [ ' . $this->getRealFullPath() . ' ] - cannot save asset',
+                    0,
+                    null
+                );
+                $duplicateFullPathException->setDuplicateElement($duplicate);
+
+                throw $duplicateFullPathException;
```

Comment:
Parameter 0 and null can be removed. This should be enough (Also for object/document):
```suggestion
                throw new DuplicateFullPathException('Duplicate full path [ ' . $this->getRealFullPath() . ' ] - cannot save asset')
                    ->setDuplicateElement($duplicate);
```